data-hibernate-jpa/src/main/java/io/micronaut/data/hibernate/operations/HibernateJpaOperations.java

```diff
@@ -152,8 +152,11 @@ protected void setParameter(CommonQueryContract query, String parameterName, Obj
 
     @Override
     protected void setParameterList(CommonQueryContract query, String parameterName, Collection<Object> value) {
+        if (value == null) {
```

Comment:
added these null checks to a few more places where a collection was guaranteed and `value` was used